proto/substrait/algebra.proto

```diff
@@ -228,6 +228,71 @@ message AggregateRel {
   }
 }
 
+// The Window relational operate on a group of rows, referred to as a window, and calculate a return value for each row based on the group of rows.
+message WindowRel {
+  RelCommon common = 1;
+  Rel input = 2;
+  repeated WindowRelFunction window_functions = 3;
```

Comment:
Both Spark & Velox have an operator which receives a single sort/partition and a collection of functions to apply given that ordering/grouping.

This is is a physical relation targeting those systems so it is ok if it is not general enough to cover all systems.

There is no logical equivalent of `WindowRel` because a logical plan will use `ProjectRel` with `WindowFunction`.  This will allow specifying partition/sort on each function.

So it sounds like Calcite could use the logical representation directly.  Though if we want to introduce a physical "window operator" (maybe calcite needs scalar and window functions to be separated?) then I would think that Calcite is welcome to do so.

Reply:
Yes. It seems the Calcite can use the logical representation. What do your think? @jacques-n 
@westonpace  Do you have any further comment for this PR? Thanks.

Reply:
@jacques-n @westonpace Can you help to review again? Thanks.